src/main/java/com/github/jcustenborder/kafka/connect/snmp/SnmpTrapSourceTask.java

```diff
@@ -280,7 +280,17 @@ private void setupMpv3Usm(Snmp snmp, SnmpTrapSourceConnectorConfig config, Secur
     USM usm = new USM(sp, new OctetString("SNMP Connector"), 0);
     usm.setEngineDiscoveryEnabled(true);
     SecurityModels sm = SecurityModels.getInstance().addSecurityModel(usm);
-    if (Utils.noneNull(config.username, config.privacyPassphrase, config.authenticationPassphrase)) {
+    if (config.noAuthNoPrivEnabled) {
+      UsmUser uu = new UsmUser(
+              new OctetString(config.username),
+              null,
+              null,
+              null,
+              null
+      );
+      usm.addUser(uu);
+      log.info("Added user {} to handle MPv3 NoAuthNoPriv", config.username);
+    } else if (Utils.noneNull(config.username, config.privacyPassphrase, config.authenticationPassphrase)) {
```

Comment:
I guess it can just be `ìf` so it can support authenticated V3 and non-authenticated. The configuration property should be enough so that it defaults to false and can be enabled or not and it should work with a proper user as well.